fault-tolerance-test/src/scaling.test.js

```diff
@@ -0,0 +1,135 @@
+const Docker = require('dockerode')
+const { sleep } = require('@jvalue/node-dry-basics')
+
+const { PIPELINE_URL, STORAGE_URL, DOCKER_COMPOSE_FILE, ENV_FILE } = require('./util/env')
+const { DockerCompose, writeDockerLogs } = require('./util/docker-compose')
+const { waitForServicesToBeReady } = require('./util/waitForServices')
+const { publishEvent } = require('./util/amqp')
+const http = require('./util/http')
+
+const SECOND = 1000
+const MINUTE = 60 * SECOND
+const TEST_TIMEOUT = 5 * MINUTE
+
+const ALL_SERVICES = [
+  'adapter',
+  'adapter-db',
+  'adapter-outboxer',
+  'edge',
+  'notification',
+  'notification-db',
+  'pipeline',
+  'pipeline-db',
+  'pipeline-outboxer',
+  'rabbitmq',
+  'scheduler',
+  'storage',
+  'storage-db',
+  // 'storage-db-liquibase', //Only creates the database schema and then exits, therefore ignore
+  'storage-mq'
+].sort()
+
+describe('Scaling test', () => {
+  let dockerCompose
+  let docker
+
+  beforeAll(() => {
+    dockerCompose = DockerCompose(DOCKER_COMPOSE_FILE, ENV_FILE)
+    docker = new Docker({})
+  })
+
+  beforeEach(async () => {
+    if ((await docker.listContainers()).length !== 0) {
+      throw Error('Can not execute restart test if other containers are running')
+    }
+
+    // Do not try to build the images because then timing is really hard
+    await dockerCompose('up -d --no-build adapter-db notification-db pipeline-db storage-db')
+    await dockerCompose('up -d --no-build storage-db-liquibase')
+    await dockerCompose('up -d --no-build adapter notification pipeline scheduler storage storage-mq')
+    await dockerCompose('up -d --no-build edge') // Do not start edge with the other services, as then traefik sometime does not get all docker events
+    await waitForServicesToBeReady()
+  }, TEST_TIMEOUT)
+
+  afterEach(async () => {
+    try {
+      await writeDockerLogs(dockerCompose, ALL_SERVICES)
+    } catch (error) {
+      console.log('Failed to save logs', error)
+    }
+
+    await dockerCompose('down')
+  }, TEST_TIMEOUT)
+
+  test('event is only consumed by one instance', async () => {
+    const datasourceId = 1
+
+    await dockerCompose('up -d --scale pipeline=2 pipeline')
```

Comment:
should we make sure actually 2 instances were started?

Reply:
Yes, because currently, the test will also pass if only one instance is running.